src/test/java/net/snowflake/client/jdbc/ResultSetMultiTimeZoneIT.java

```diff
@@ -182,6 +177,64 @@ public void testGetDateAndTime() throws SQLException {
     connection.close();
   }
 
+  /**
+   * Test for getDate(int columnIndex, Calendar cal) function to ensure it matches values with
+   * getTimestamp function
+   */
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testDateAndTimestampWithTimezone() throws SQLException {
+    Connection connection = init();
+    Statement statement = connection.createStatement();
+    statement.execute("alter session set JDBC_FORMAT_DATE_WITH_TIMEZONE=true");
+    ResultSet rs =
+        statement.executeQuery(
+            "SELECT DATE '1970-01-02 00:00:00' as datefield, "
+                + "TIMESTAMP '1970-01-02 00:00:00' as timestampfield");
+    rs.next();
+
+    // Set a timezone for results to be returned in and set a format for date and timestamp objects
+    Calendar cal = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+    SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    sdf.setTimeZone(cal.getTimeZone());
+
+    // Date object and calendar object should return the same timezone offset with calendar
+    Date dateWithZone = rs.getDate(1, cal);
+    Timestamp timestampWithZone = rs.getTimestamp(2, cal);
+    assertEquals(sdf.format(dateWithZone), sdf.format(timestampWithZone));
+
+    // When fetching Date object with getTimestamp versus Timestamp object with getTimestamp,
+    // results should match
+    assertEquals(rs.getTimestamp(1, cal), rs.getTimestamp(2, cal));
+
+    // When fetching Timestamp object with getDate versus Date object with getDate, results should
+    // match
+    assertEquals(rs.getDate(1, cal), rs.getDate(2, cal));
+
+    // getDate() without Calendar offset called on Date type should return the same date with no
+    // timezone offset
+    assertEquals("1970-01-02 00:00:00", sdf.format(rs.getDate(1)));
+    // getDate() without Calendar offset called on Timestamp type returns date with timezone offset
+    assertEquals("1970-01-02 08:00:00", sdf.format(rs.getDate(2)));
+
+    // getTimestamp() without Calendar offset called on Timestamp type should return the timezone
+    // offset
+    assertEquals("1970-01-02 08:00:00", sdf.format(rs.getTimestamp(2)));
+    // getTimestamp() without Calendar offset called on Date type should not return the timezone
+    // offset
+    assertEquals("1970-01-02 00:00:00", sdf.format(rs.getTimestamp(1)));
+
+    // test that session parameter functions as expected. When false, getDate() has same behavior
+    // with or without
+    // Calendar input
+    statement.execute("alter session set JDBC_FORMAT_DATE_WITH_TIMEZONE=false");
```

Comment:
can you add the test cases when `JDBC_FORMAT_DATE_WITH_TIMEZONE` is false? Those should match the current behaviors to guarantee no regression.

Reply:
I have the 1 statement below (assertEquals(rs.getDate(1, cal), rs.getDate(1));) but I could add more

Reply:
Ok, you think it is good enough? Then that's fine.

Reply:
I added a hard-coded value to check against just to be more sure.